pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -6598,6 +6598,99 @@ spec:
 	}
 }
 
+func TestReconcile_ParamsValidationsImmediatelyFailPipelineRun(t *testing.T) {
+	names.TestingSeed()
+
+	ctx := context.Background()
+	cfg := config.NewStore(logtesting.TestLogger(t))
+	ctx = cfg.ToContext(ctx)
+
+	prs := []*v1beta1.PipelineRun{
+		parse.MustParseV1beta1PipelineRun(t, `
+metadata:
+  name: pipelinerun-matrix-param-invalid-type
+  namespace: foo
+spec:
+  pipelineSpec:
+   tasks:
+    - name: mytask
+      taskSpec:
+        steps:
+         - name: echo
+           image: alpine
+           script: |
+             echo "test"
+    - name: platforms-and-browsers
+      taskRef:
+        name: mytask
+      matrix:
+        params:
+          - name: platform
+            value: linux
+          - name: browser
+            value:
+              - chrome
+              - safari
+              - firefox
+      params:
+        - name: version
+          value: v0.33.0
+`),
+		parse.MustParseV1beta1PipelineRun(t, `
+metadata:
+  name: pipelinerun-matrix-param-duplicates
+  namespace: foo
+spec:
+  pipelineSpec:
+   tasks:
+    - name: mytask
+      matrix:
+        params:
+          - name: browser
+            value:
+              - chrome
+          - name: browser
+            value:
+              - safari
+              - firefox
+      taskSpec:
+        steps:
+         - name: echo
+           image: alpine
+           script: |
+             echo "test"
+`),
+	}
+
+	cms := []*corev1.ConfigMap{withEnabledAlphaAPIFields(newFeatureFlagsConfigMap())}
+	d := test.Data{
+		PipelineRuns: prs,
+		ConfigMaps:   cms,
+	}
+	prt := newPipelineRunTest(t, d)
+	defer prt.Cancel()
+
+	run1, _ := prt.reconcileRun("foo", "pipelinerun-matrix-param-invalid-type", nil, true)
```

Comment:
can you refactor to use table-driven testing here?
https://github.com/tektoncd/community/blob/main/standards.md#tests: "Where possible, each test should have a focused set of inputs and outputs. For example, only one function and set of inputs should be tested at a time, the set of inputs should be limited to make it clear what's being tested"

Reply:
Combined with the other existing failure tests

Reply:
What I had in mind here was more about separating different inputs and outputs into different test cases; i.e. instead of run1, run2, run3, run4 having 4 different test cases that each have their own run. I know some of this logic is already present which is unfortunate but I'd rather not add to it

Reply:
Okay, refactored these into separate test cases.